test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -292,6 +269,45 @@ public void beforeEach(ExtensionContext context) throws Exception {
         }
     }
 
+    @Override
+    public void afterEach(ExtensionContext context) throws Exception {
+        if (isNativeTest(context)) {
+            return;
+        }
+        if (!failedBoot) {
+            popMockContext();
+            for (Object afterEachCallback : afterEachCallbacks) {
+                Map.Entry<Class<?>, ?> tuple = createQuarkusTestMethodContextTuple(context);
+                afterEachCallback.getClass().getMethod("afterEach", tuple.getKey())
+                        .invoke(afterEachCallback, tuple.getValue());
+            }
+            boolean nativeImageTest = isNativeTest(context);
```

Comment:
Isn't this going to be always false?

Reply:
That must be leftover - I just moved the method. I'll take a look tomorrow

Reply:
Good catch, removed